### PR TITLE
[Improve][Connector-V2][My Hours]Add http method enum && Improve My Hours connector option rule

### PR DESCRIPTION
**Pull request by TaoZex**
## Purpose of this pull request

1.Add http method enum

320ee1850f1caeeda7502ba8422bfc1

2.Improve My Hours connector option rule.

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/incubator-seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/incubator-seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-dist/pom.xml)

### CODE REVIEW

File: seatunnel-connectors-v2/connector-http/connector-http-myhours/src/main/java/org/apache/seatunnel/connectors/seatunnel/myhours/source/MyHoursSourceFactory.java [hunk 1]
```diff
@@ -37,6 +40,13 @@ public OptionRule optionRule() {
                 .required(MyHoursSourceConfig.URL)
                 .required(MyHoursSourceConfig.EMAIL)
                 .required(MyHoursSourceConfig.PASSWORD)
+                .optional(MyHoursSourceConfig.METHOD)
+                .optional(MyHoursSourceConfig.HEADERS)
+                .optional(MyHoursSourceConfig.PARAMS)
+                .conditional(Condition.of(HttpConfig.METHOD, "post"), MyHoursSourceConfig.BODY)
```
Comment (EricJoy2048) [inline seatunnel-connectors-v2/connector-http/connector-http-myhours/src/main/java/org/apache/seatunnel/connectors/seatunnel/myhours/source/MyHoursSourceFactory.java]: Please use enum type option.

Comment (TyrantLucifer) [reply seatunnel-connectors-v2/connector-http/connector-http-myhours/src/main/java/org/apache/seatunnel/connectors/seatunnel/myhours/source/MyHoursSourceFactory.java]: +1, please use enum type option.

File: seatunnel-connectors-v2/connector-http/connector-http-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/http/config/Method.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.seatunnel.connectors.seatunnel.http.config;
+
+public enum Method {
```
Comment (TyrantLucifer) [inline seatunnel-connectors-v2/connector-http/connector-http-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/http/config/Method.java]: `HttpRequestMethod` is better.

File: seatunnel-connectors-v2/connector-http/connector-http-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/http/config/HttpParameter.java [hunk 3]
```diff
@@ -43,9 +43,10 @@ public void buildWithConfig(Config pluginConfig) {
         this.setUrl(pluginConfig.getString(HttpConfig.URL.key()));
         // set method
         if (pluginConfig.hasPath(HttpConfig.METHOD.key())) {
-            this.setMethod(pluginConfig.getString(HttpConfig.METHOD.key()));
+            Method method = Method.valueOf(pluginConfig.getString(HttpConfig.METHOD.key()).toUpperCase());
+            this.setMethod(method);
         } else {
-            this.setMethod(HttpConfig.METHOD_DEFAULT_VALUE);
+            this.setMethod(Method.GET);
```
Comment (TyrantLucifer) [inline seatunnel-connectors-v2/connector-http/connector-http-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/http/config/HttpParameter.java]: `HttpConfig.METHOD.defaultValue()` is better.

Comment (TaoZex) [reply seatunnel-connectors-v2/connector-http/connector-http-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/http/config/HttpParameter.java]: Thanks. I have fixed it.

**Comment by TaoZex** [comment]
@EricJoy2048 @Hisoka-X PTAL

File: seatunnel-connectors-v2/connector-http/connector-http-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/http/config/HttpConfig.java [hunk 4]
```diff
@@ -23,17 +23,16 @@
...
     public static final int DEFAULT_RETRY_BACKOFF_MAX_MS = 10000;
     public static final Option<String> URL = Options.key("url")
             .stringType()
             .noDefaultValue()
             .withDescription("Http request url");
-    public static final Option<String> METHOD = Options.key("method")
-            .stringType()
-            .defaultValue(METHOD_DEFAULT_VALUE)
+    public static final Option<HttpRequestMethod> METHOD = Options.key("method")
+            .objectType(HttpRequestMethod.class)
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/connector-http/connector-http-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/http/config/HttpConfig.java]: please use `enumType`

Comment (TaoZex) [reply seatunnel-connectors-v2/connector-http/connector-http-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/http/config/HttpConfig.java]: Thanks for your advice.I have fixed it.

**Review by Hisoka-X — APPROVED** [review]

**Review by EricJoy2048 — APPROVED** [review]

**Pull request merged**